statsmodels/tsa/arima/estimators/gls.py

```diff
@@ -240,6 +240,16 @@ def gls(endog, exog=None, order=(0, 0, 0), seasonal_order=(0, 0, 0, 0),
         # argument applies a Prais-Winsten-type normalization to the first few
         # observations to ensure homoskedasticity). Brockwell and Davis
         # mention that they also take this approach in practice.
+
+        # GH-6540: AR must be stationary
+
+        if ar_params.shape[0] and \
+                np.max(np.abs(np.roots(np.r_[1, -ar_params]))) >= 1.0:
+            raise ValueError(
+                "Roots of the autoregressive parameters indicate that data is"
+                "non-stationary. GLS cannot be used with non-stationary "
+                "parameters."
```

Comment:
I think that this will typically only happen with Hannan-Rissanen estimation, since I think the other methods all enforce stationarity (of course, up to some numerical precision).  What do you think about something sort of like the following?

```python
if not p_arma.is_stationary:
    msg = ("Roots of the autoregressive parameters indicate that data"
           " is non-stationary. GLS cannot be used with non-stationary"
           " parameters. One option is to increase the differencing"
           " order used in the model.")
    if arma_estimator == 'hannan_rissanen':
        msg += (' Alternatively, you can try a different'
                ' `arma_estimator` method (not "hannan_rissanen").')
    raise ValueError(msg)
```

Reply:
I have added this, and also mentioned nonlinear transformation such as ln.